packages/plugin-assets-retry/src/runtime/asyncChunkRetry.ts

```diff
@@ -0,0 +1,239 @@
+      originalScriptFilename,
+    // + getUrlRetryQuery(existRetryTimes),
+
+    originalScriptFilename,
+    originalSrcUrl,
+  };
+}
+
+function nextRetry(chunkId: string): Retry {
+  const currRetry = getCurrentRetry(chunkId);
+
+  let nextRetry: Retry;
+  if (!currRetry) {
+    nextRetry = initRetry(chunkId);
+  } else {
+    const existRetryTimes = currRetry.existRetryTimes + 1;
+    const nextDomain = findNextDomain(currRetry.nextDomain);
+    nextRetry = {
+      existRetryTimes,
+      nextDomain,
+      nextRetryUrl:
+        nextDomain +
+        (nextDomain.endsWith('/') ? '' : '/') +
+        currRetry.originalScriptFilename,
+      // + getUrlRetryQuery(existRetryTimes),
+
+      originalScriptFilename: currRetry.originalScriptFilename,
+      originalSrcUrl: currRetry.originalSrcUrl,
+    };
+  }
+
+  retryCollector[chunkId] = nextRetry;
+  return nextRetry;
+}
+
+// rewrite webpack runtime with nextRetry()
+const originalEnsureChunk = __RUNTIME_GLOBALS_ENSURE_CHUNK__;
+const originalGetChunkScriptFilename =
+  __RUNTIME_GLOBALS_GET_CHUNK_SCRIPT_FILENAME__;
+const originalLoadScript = __RUNTIME_GLOBALS_LOAD_SCRIPT__;
+
+// if users want to support es5, add Promise polyfill first https://github.com/webpack/webpack/issues/12877
+function ensureChunk(chunkId: string): Promise<unknown> {
+  const result = originalEnsureChunk(chunkId);
+  const originalScriptFilename = originalGetChunkScriptFilename(chunkId);
+
+  // mark the async chunk name in the global variables and share it with initial chunk retry
+  if (window && !window.__ASYNC_CHUNK_FILENAME_LIST__[originalScriptFilename]) {
+    window.__ASYNC_CHUNK_FILENAME_LIST__[originalScriptFilename] = true;
+  }
+
+  return result.catch(function (error: Error) {
+    const { existRetryTimes, originalSrcUrl, nextRetryUrl, nextDomain } =
+      nextRetry(chunkId);
+
+    if (existRetryTimes > maxRetries) {
+      error.message = `Loading chunk ${chunkId} from ${originalSrcUrl} failed after ${maxRetries} retries.`;
+      throw error;
+    }
+
+    // Filter by config.test and config.domain
+    let tester = config.test;
+    if (tester) {
+      if (typeof tester === 'string') {
+        const regexp = new RegExp(tester);
+        tester = (str: string) => regexp.test(str);
+      }
+
+      if (typeof tester !== 'function' || !tester(nextRetryUrl)) {
+        throw error;
+      }
+    }
+
+    if (
+      config.domain &&
+      config.domain.length > 0 &&
+      config.domain.indexOf(nextDomain) === -1
+    ) {
+      throw error;
+    }
+
+    const context: AssetsRetryHookContext = {
+      times: existRetryTimes,
+      domain: nextDomain,
+      url: nextRetryUrl,
+      tagName: 'script',
+    };
+
+    if (config.onRetry && typeof config.onRetry === 'function') {
+      config.onRetry(context);
+    }
+
+    // biome-ignore lint/complexity/useArrowFunction: use function instead of () => {}
+    return new Promise(function (resolve) {
+      // TODO: options to set retryDelay
```

Comment:
when do user need to set a retryDelay

Reply:
I think almost all the error retry of requests will set a retryInterval option,  assets-retry may also need one

https://swr.vercel.app/docs/error-handling.en-US#error-retry
https://ahooks.js.org/hooks/use-request/retry#error-retry